src/long.ts

```diff
@@ -246,13 +247,43 @@ export class Long extends BSONValue {
   }
 
   /**
+   * @internal
+   * Returns false for an string that contains invalid characters for its radix, else returns the original string.
+   * @param str - The textual representation of the Long
+   * @param radix - The radix in which the text is written (2-36), defaults to 10
+   */
+  static validateStringCharacters(str: string, radix?: number): false | string {
+    radix = radix ?? 10;
+
+    let regexInputString = '';
+    if (radix <= 10) {
+      regexInputString = `[^-0-${radix - 1}+]`;
+    } else {
+      const validCharRangeEnd = String.fromCharCode('a'.charCodeAt(0) + (radix - 11));
+      regexInputString = `[^-0-9+(a-${validCharRangeEnd})]`;
+    }
+    const regex = new RegExp(regexInputString, 'i');
+    return regex.test(str) ? false : str;
+  }
+
+  /**
+   * @internal
    * Returns a Long representation of the given string, written using the specified radix.
+   * Throws an error if `throwsError` is set to true and any of the following conditions are true:
+   *  - the string contains invalid characters for the given radix
+   *  - the string contains whitespace
    * @param str - The textual representation of the Long
    * @param unsigned - Whether unsigned or not, defaults to signed
    * @param radix - The radix in which the text is written (2-36), defaults to 10
+   * @param throwsError - Whether or not throwing an error is permitted
    * @returns The corresponding Long value
    */
-  static fromString(str: string, unsigned?: boolean, radix?: number): Long {
+  static fromStringHelper(
+    str: string,
+    unsigned?: boolean,
+    radix?: number,
+    throwsError?: boolean
```

Comment:
Generally, options are named imperatively, not passively.  so this might be `throwError` instead of `throwsError`.  

I'd also suggest making the name more descriptive - it's not accurate to say this option controls whether errors are thrown or not. When `throwsError` is false, we throw errors if the string is empty or in other invalid cases.  This option specifically controls whether or not we validate characters in the string, so a name such as `validateStringCharacters` might be more appropriate.

Also, it's best practice to only make values optional when they're sometimes not provided.  In this PR, you call `fromStringHelper` in two places, both of which provide a value for `throwsError`.

Lastly, fromStringHelper isn't very descriptive.  We've [used the pattern of a shared internal method in the past](https://github.com/mongodb/node-mongodb-native/blob/eece8c1b9fcd0aa7c12722f4fe4e50e2123102b8/src/client-side-encryption/client_encryption.ts#L677) when we need two different public APIs with overlapping implementations and we chose the naming convention `private _<methodName>` to indicate that it is internal.  In this case, `private _fromString(...)` instead of `fromStringHelper`.

Reply:
yep, sounds good made these changes :)